src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyle.cs

```diff
@@ -311,6 +337,31 @@ private bool CheckAllLower(string name, TextSpan nameSpan, out string reason)
                 first = false;
             }
 
+            if (!violations.Any() && name.Contains("_"))
+            {
+                var rx = new Regex("(?<=[_])");
+                var words = rx.Split(name).Where(str => str.Length > 0);
+                var spanStart = 0;
+                var firstWord = true;
+                foreach (var word in words)
+                {
+                    if (firstWord)
+                    {
+                        firstWord = false;
+                        spanStart += word.Length;
+                        continue;
+                    }
+
+                    var spanToCheck = TextSpan.FromBounds(spanStart, spanStart + word.Length);
+                    if (!restWordCheck(name, spanToCheck))
+                    {
+                        violations.Add(Substring(name, spanToCheck));
+                    }
+
+                    spanStart += word.Length;
+                }
+            }
+
```

Comment:
I wonder if you could turn this into a function that takes in `restWordCheck` as a parameter instead? It is identical to the function defined above except the other one calls `wordCheck`.